contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -4971,6 +4978,7 @@ transform_pivot_clause(ParseState *pstate, SelectStmt *stmt)
 	/* use the copy of the orgininal fromClause to prevent double analyzing fromClause */
 	((SelectStmt *)stmt->srcSql)->fromClause = src_sql_fromClause_copy;
 	/* Transform the new src_sql & get the output type of that agg function*/
+	((SelectStmt *)stmt->srcSql)->withClause = with_clause;
```

Comment:
If I understand correctly we are finally placing pivot_src_sql as the source sql in bbf_pivot_function.
could you point where pivot_src_sql-> fromClause & withClause go through transformation ?

Reply:
If pivot_src_sql is a select stmt with pivot operator, then it will go through transform_pivot_clause hook again. If there is no more pivot in subquery,  pivot_src_sql is just a normal select stmt, and it will go through the original fromClause and withClause transformation.